src/Propel/Runtime/ActiveQuery/Criteria.php

```diff
@@ -1063,15 +1063,17 @@ public function addJoin($left, $right, ?string $joinType = null)
         $join->setIdentifierQuoting($this->isIdentifierQuotingEnabled());
 
         // is the left table an alias ?
-        $dotpos = strrpos($left, '.');
-        $leftTableAlias = substr($left, 0, $dotpos);
-        $leftColumnName = substr($left, $dotpos + 1);
+        /** @phpstan-var string $left */
+        $length = strrpos($left, '.') ?: null;
```

Comment:
Why do you prefer `$length` over `$dotpos`? The latter seems more declarative?

Reply:
$dotPos sounds fair

Reply:
The result is used as `$length` for substr method to retrieve the leftTableAlias, as well to get `$offset` to retrieve the leftColumnName which is which is depending on the `$length` used before.

Reply:
Yes, that I can see. But length of what though? I get that `$dotpos` is a weird abbrev, even coming from a function named `strpos`. But at least I can infer that it is the position of a dot. The name `$length` tells me nothing, except that it is some point in an unknown dimension. Shouldn't a rename rather be more precise than less, i.e. `$firstDotPosition` or `$seperatorIndex` or such?

Reply:
I would also say, that the length name of the target method is also coming to a point later in code.
So here we don't know that yet, so the name usually (also in other pieces of the code) is named after what is done here, the position finding.